x-pack/plugins/security_solution/server/endpoint/lib/artifacts/task.ts

```diff
@@ -89,37 +90,66 @@ export class ManifestTask {
       return;
     }
 
-    let errors: Error[] = [];
     try {
-      // get snapshot based on exception-list-agnostic SOs
-      // with diffs from last dispatched manifest
-      const snapshot = await manifestManager.getSnapshot();
-      if (snapshot && snapshot.diffs.length > 0) {
-        // create new artifacts
-        errors = await manifestManager.syncArtifacts(snapshot, 'add');
-        if (errors.length) {
-          reportErrors(this.logger, errors);
-          throw new Error('Error writing new artifacts.');
-        }
-        // write to ingest-manager package config
-        errors = await manifestManager.dispatch(snapshot.manifest);
-        if (errors.length) {
-          reportErrors(this.logger, errors);
-          throw new Error('Error dispatching manifest.');
+      // Last manifest we computed, which was saved to ES
+      const oldManifest = await manifestManager.getLastComputedManifest(
+        ManifestConstants.SCHEMA_VERSION
+      );
+      if (oldManifest == null) {
+        this.logger.debug('User manifest not available yet.');
+        return;
+      }
+
+      // New computed manifest based on current state of exception list
+      const newManifest = await manifestManager.buildNewManifest(
+        ManifestConstants.SCHEMA_VERSION,
+        oldManifest
+      );
+      const diffs = newManifest.diff(oldManifest);
+
+      // Compress new artifacts
+      const adds = diffs.filter((diff) => diff.type === 'add').map((diff) => diff.id);
+      for (const artifactId of adds) {
+        const compressError = await newManifest.compressArtifact(artifactId);
+        if (compressError) {
+          throw compressError;
         }
-        // commit latest manifest state to user-artifact-manifest SO
-        const error = await manifestManager.commit(snapshot.manifest);
+      }
+
+      // Persist new artifacts
+      const artifacts = adds
+        .map((artifactId) => newManifest.getArtifact(artifactId))
+        .filter((artifact) => artifact !== undefined) as InternalArtifactSchema[];
```

Comment:
Similar comment as above related to using `as`. Could you make it work by adding the type in the parameters of `.map()` and `.filter()`?